CMakeLists.txt

```diff
@@ -176,3 +176,5 @@ target_link_libraries(sof PRIVATE sof_static_libraries)
 sof_add_build_counter_rule()
 
 add_subdirectory(src)
+
+sof_append_relative_path_definitions(sof)
```

Comment:
I believe, that this line should be duplicated for host builds, inside `if(BUILD_LIBRARY)` after `add_subdirectory(src)` at line 127. That should solve host build issues.

Reply:
Probably yea. It's a bit painful to use this function for every target that uses this define, but it's the best what I could get with limitations of CMake. I tried to set up it globally with taking all libs recursively and their sources but CMake can set source file property only IRT targets defined in the same CMakeLists, so there were no gain. Thought also about adding it to add_local_sources, but sometimes people put sources directly fe in add_executable so it's also meh. Maybe you have some idea?

Reply:
Lets consider and fix this is a subsequent PR, the reproducability is prime.